[windows] conditionally add windows cws components.

derekwbrown opened the pull request:
### What does this PR do?


Adds Windows CWS agent to windows installer conditionally.

Components are added only based on inclusion in `release.json`.
Components are included in nightly builds, but not in release builds

### Motivation

Include under development components in nightly builds to facilitate ongoing development without including in release builds

### Additional Notes


### Possible Drawbacks / Trade-offs

### Describe how to test/QA your changes


### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

derekwbrown committed:
[windows] conditionally add windows cws components.

If configured in `release.json`, add the windows components
- windows security agent
- procmon merge module

clarkb7 commented:
does `security-agent.exe` need to be added to `additional_sign_files_list` so it gets signed?

derekwbrown committed:
Add proper signing for newly created security-agent.exe

derekwbrown committed:
make the windows probe object auto-start the procmon driver

chouquette commented on omnibus/config/projects/agent.rb:
```diff
@@ -230,6 +230,12 @@
   end
 
+  include_procmon = "false"
+  if not windows_arch_i386? and ENV['WINDOWS_DDPROCMON_DRIVER'] and not ENV['WINDOWS_DDPROCMON_DRIVER'].empty?
```
:question: not directly related to this PR, but do we still support Windows 32bits?

chouquette replied:
Couldn't you check if the file exists rather than probing the environment? If I understand your change correctly, `security-agent.exe` will only be there if it was built, based on the same condition in `datadog-agent.rb`
This would be simpler to read and to toggle IMO

chouquette commented on omnibus/config/projects/agent.rb:
```diff
@@ -411,6 +418,10 @@ def generate_embedded_archive(version)
   windows_symbol_stripping_file "#{Omnibus::Config.source_dir()}\\datadog-agent\\src\\github.com\\DataDog\\datadog-agent\\bin\\agent\\trace-agent.exe"
   windows_symbol_stripping_file "#{Omnibus::Config.source_dir()}\\datadog-agent\\src\\github.com\\DataDog\\datadog-agent\\bin\\agent\\agent.exe"
   windows_symbol_stripping_file "#{Omnibus::Config.source_dir()}\\datadog-agent\\src\\github.com\\DataDog\\datadog-agent\\bin\\agent\\system-probe.exe"
+  if not windows_arch_i386? and ENV['WINDOWS_DDPROCMON_DRIVER'] and not ENV['WINDOWS_DDPROCMON_DRIVER'].empty?
```
Same question about checking the file existence rather than the flag

chouquette commented on omnibus/config/software/datadog-agent-finalize.rb:
```diff
@@ -28,6 +28,9 @@
             if ENV['WINDOWS_DDNPM_DRIVER'] and not ENV['WINDOWS_DDNPM_DRIVER'].empty? and not windows_arch_i386?
               move "#{install_dir}/etc/datadog-agent/system-probe.yaml.example", conf_dir_root, :force=>true
             end
+            if ENV['WINDOWS_DDPROCMON_DRIVER'] and not ENV['WINDOWS_DDPROCMON_DRIVER'].empty? and not windows_arch_i386?
```
Same question

chouquette reviewed (approved):
LGTM for agent-platform files, with non-blocking comments

derekwbrown committed:
update to release signed

derekwbrown committed:
ADd windows specific outlines to security config tamplate

Add windows specific defaults to system probe security config
Split entries in config that are platform specific into separate files

paulcacheux commented on pkg/config/system_probe_cws_windows.go:
```diff
@@ -0,0 +1,20 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Can you add this file and the `notwin` version to the CODEOWNERS file (owned by our team) ? So that it matches the rest of our config

derekwbrown replied:
yup.  And renamed the confusing named `windows_notwin.go` file too

derekwbrown committed:
rename misnamed file; update codeowners

derekwbrown committed:
gofmt

buraizu commented on pkg/config/config_template.yaml:
```diff
@@ -1912,22 +1912,35 @@ api_key:
+{{- if (eq .OS "windows")}}
+    ## @param dir - string - default: %ProgramData%\Datadog\runtime-security.d
+    ## @env DD_RUNTIME_SECURITY_CONFIG_POLICIES_DIR - string - default: /etc/datadog-agent/runtime-security.d
+    ## Path from where the policy files will be loaded
```
```suggestion
    ## Path from where the policy files are loaded
```

buraizu reviewed (approved):
Thanks, approving with a minor suggestion

derekwbrown committed:
Update pkg/config/config_template.yaml

Co-authored-by: Bryce Eadie <bryce.eadie@datadoghq.com>

derekwbrown committed:
Review feedback.

Make the installer stop the newly added services when present.

derekwbrown committed:
remove duplicate line

derekwbrown committed:
fix missing ;

dustmop commented on pkg/config/config_template.yaml:
```diff
@@ -1912,22 +1912,35 @@ api_key:
+  ## @env DD_SYSTEM_PROBE_CONFIG_SYSPROBE_SOCKET - string - optional - default: localhost:3334
+  ## The TCP address where the security runtime module is accessed.
+  #
+  # socket: localhost:3334
```
The `@param` above is `sysprobe_socket`, which doesn't match here or the usages in `pkg/config/system_probe_cws_windows.go`. Which is correct?

derekwbrown replied:
sysprobe socket & socket are different; socket is for the cws component.
However, I have a note (for a separate pr) that the system probe config needs some windows-specific updating.

dustmop reviewed (approved):
LGTM for agent-shared-components, with one question
